src/world.js

```diff
@@ -156,7 +156,33 @@ class World
 
   dfs()
   {
-    // TODO
+    
+
+    let n = this.frontier.length;
+    for (let i = 0; !this.goalFound && i < n; i++)
```

Comment:
This section should look something like this:

```js
dfs()
{
  let n = this.frontier.length;

  for (let i = 0; !this.goalFound && i < 4; i++)
  {
    let current = this.frontier.pop();

    if (current == this.goal)
    {
      this.goalFound = true;
      this.setPath();
    }

    for (let next of this.neighbors(current))
    {
      if (!this.reached.has(next))
      {
        this.frontier.push(next);
        this.reached.add(next);
        this.cameFrom.set(next, current);
      }
    }
  }
}
```

I've limited the number of iterations to 4, which improves the algorithm's visualization. I've also changed a lit bit of the code's formatting to make it more consistent with the rest of it.